pkg/http/proxy.go

```diff
@@ -33,10 +33,10 @@ const NoHostOverride = ""
 // If hostOverride is not an empty string, the outgoing request's Host header will be
 // replaced with that explicit value and the passthrough loadbalancing header will be
 // set to enable pod-addressability.
-func NewHeaderPruningReverseProxy(target, hostOverride string, headersToRemove []string) *httputil.ReverseProxy {
+func NewHeaderPruningReverseProxy(target, hostOverride string, headersToRemove []string, scheme string) *httputil.ReverseProxy {
```

Comment:
Elsewhere we've used a `bool` for TLS or not. Do we intend this function to support protocols other than `http` and `https`?